photon_stream/JsonLinesReader.py

```diff
@@ -0,0 +1,39 @@
+import ujson as json
+import os
+import gzip
+
+
+class JsonLinesReader:
+    """
+    Sequentially reads a JSON-Lines file line by line and provides
+    dictionaries for each line. Also supports gzipped files.
+    """
+    def __init__(self, path):
+        self.path = os.path.abspath(path)
+
+        # check for gzip file
+        with open(path, 'rb') as f:
+            marker = f.read(2)
+            gzipped = marker[0] == 31 and marker[1] == 139
```

Comment:
Maybe add a link to https://en.wikipedia.org/wiki/Gzip or even https://tools.ietf.org/html/rfc1952#page-5 in the comment.